Registration/RemoteManagement.psm1

```diff
@@ -0,0 +1,489 @@
+﻿# Copyright (c) Microsoft Corporation. All rights reserved.
+# See LICENSE.txt in the project root for license information.
+
+<#
+This module contains functions for creating edge subscription resource for default provider subscription and enabling remote management on AzureStack. 
+This is supported only for connected AzureStack scenarios. You must use same Azure credentials that are used for registration.
+#>
+
+<#
+
+.SYNOPSIS
+
+Registers the AzureStack resource provider in this environment
+
+#>
+function Register-AzureStackResourceProvider{
+[CmdletBinding()]
+
+    $currentAttempt = 0
+    $maxAttempt = 3
+    $sleepSeconds = 10
+    do
+    {
+        try
+        {
+            Log-Output "Registering Azure Stack resource provider."
+            [Version]$azurePSVersion = (Get-Module AzureRm.Resources).Version
+            if ($azurePSVersion -ge [Version]"4.3.2")
```

Comment:
you can do(gcm Register-AzureRmResourceProvider).Parameters.ContainsKey('Force')) instead of version check

Reply:
this is same as registration code. Like to reuse existing implementation